crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,285 @@
+/// for details
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize, Default)]
+pub enum StorageMode {
+    /// Monitor for `Notes` related to the account, in addition to being `Public`.
+    #[serde(alias = "network")]
+    #[default]
+    Network,
+    /// A publicly stored account, lives on-chain.
+    #[serde(alias = "public")]
+    Public,
+    /// A private account, which must be known by interactors.
+    #[serde(alias = "private")]
+    Private,
+}
+
+impl From<StorageMode> for AccountStorageMode {
+    fn from(mode: StorageMode) -> AccountStorageMode {
+        match mode {
+            StorageMode::Network => AccountStorageMode::Network,
+            StorageMode::Private => AccountStorageMode::Private,
+            StorageMode::Public => AccountStorageMode::Public,
+        }
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum AssetKind {
+    Fungible,
+    NonFungible,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct AssetEntry {
+    symbol: String, // TODO move to a wrapper around `TokenSymbol`
+    amount: u64,    // TODO we might want to provide `humantime`-like denominations
+}
+
+#[allow(missing_docs, reason = "Error variants must be descriptive by themselves")]
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Toml(#[from] toml::de::Error),
+    #[error("Account translation from config to state failed")]
+    Account(#[from] AccountError),
+    #[error("Asset translation from config to state failed")]
+    Asset(#[from] AssetError),
+    #[error("Applying assets to account failed")]
+    AccountDelta(#[from] miden_objects::AccountDeltaError),
+    #[error("You defined an asset {symbol:?} that has no faucet producing it")]
+    MissingFaucetDefinition { symbol: TokenSymbol },
+    #[error(transparent)]
+    TokenSymbol(#[from] TokenSymbolError),
+    #[error("The provided max supply {max_supply} exceeds the field modulus {modulus}")]
+    MaxSupplyExceedsFieldModulus { max_supply: u64, modulus: u64 },
+    #[error("Unsupported value for key {key} : {value}")]
+    UnsupportedValue {
+        key: &'static str,
+        value: String,
+        message: String,
+    },
+    #[error("Failed to create fungible faucet account")]
+    FungibleFaucet(#[from] FungibleFaucetError),
+}
+
+/// Secrets generated during the state generation
+///
+/// Attention: Only to be used for testing, for all other usecases
+/// inject only the public keys.
+#[derive(Debug, Clone)]
+pub struct TestSecrets {
+    pub secrets: Vec<(AccountType, AccountId, SecretKey)>,
+}
+
+/// Specify a set of faucets and wallets with assets for easier test depoyments
+///
+/// Notice: Any faucet must be declared _before_ it's use in a wallet/regular account.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct TestGenesisConfig {
```

Comment:
WDYT about `GenesisConfig` and then we use it ourselves as well?

Users will need some way of calling this from the node binary, so we already want to add it to the CLI I think (unless I'm misunderstanding the use case @mmagician?).

If so, then we'll want to add it to the bundled `start` command as an optional path to the toml file. And if its missing then have our current config as the default.